src/server/service/diagnostics/log.rs

```diff
@@ -231,11 +246,28 @@ fn parse_level(input: &str) -> IResult<&str, &str> {
 }
 
 /// Parses the single log line and retrieve the log meta and log body.
-fn parse(input: &str) -> IResult<&str, (i64, LogLevel)> {
-    let (content, (time, level, _)) = tuple((parse_time, parse_level, space1))(input)?;
-    let timestamp = match DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z") {
-        Ok(t) => t.timestamp_millis(),
-        Err(_) => INVALID_TIMESTAMP,
+fn parse(input: &str) -> Result<(&str, (i64, LogLevel)), Error> {
+    let content;
```

Comment:
We can make the code more "rustic":
```rust
let (content, (time, level, _)) = tuple((parse_time, parse_level, space1))(input).map_err(|err| Err(Error::ParseError(err.to_string()))?;
let timestamp = DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z").map_err(|err| Err(Error::ParseError(err.to_string())))?.timestamp_millis();
```

Reply:
cool! much simpler now, good to know how map_err works, will update it soon.

Reply:
I'm trying on this, but it says:

```
 the trait `std::convert::From<std::result::Result<_, server::service::diagnostics::log::Error>>` is not implemented for `server::service::diagnostics::log::Error`
```

I am still trying to learn and resolving it.

Reply:
```rust
    let (content, (time, level, _)) = tuple((parse_time, parse_level, space1))(input)
        .map_err(|err| Error::ParseError(err.to_string()))?;
    let timestamp = DateTime::parse_from_str(time, "%Y/%m/%d %H:%M:%S%.3f %z")
        .map_err(|err| Error::ParseError(err.to_string()))?
        .timestamp_millis();
```
sorry, I make a mistake

Reply:
Got it! it works now, thanks!